frontend/packages/ceph-storage-plugin/src/components/kms-config/hpcs-config.tsx

```diff
@@ -87,7 +61,7 @@ export const HpcsConfigure: React.FC<KMSConfigureProps> = ({ state, dispatch, cl
       >
         <TextInput
           value={kms.name?.value}
-          onChange={setServiceName}
+          onChange={setRequiredParams('name')}
```

Comment:
You converted into a single generic function that's good, but it introduced the risk of maintaining key names in multiple places.

A generic function is feasible only when enough constants and enum are shared between callers. Either you need to convert all HPCS keys into enum/constants and reuse them in this function call and the object declaration (or)  the old approach i.e individual functions for each key.